lib/node_modules/@stdlib/math/base/special/log2f/lib/main.js

```diff
@@ -0,0 +1,149 @@
+*
+* ```text
+* Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+*
+* Developed at SunPro, a Sun Microsystems, Inc. business.
+* Permission to use, copy, modify, and distribute this
+* software is freely granted, provided that this notice
+* is preserved.
+* ```
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
+var f32 = require( '@stdlib/number/float64/base/to-float32' );
+var toWordf = require( '@stdlib/number/float32/base/to-word' );
+var fromWordf = require( '@stdlib/number/float32/base/from-word' );
+var ABS_MASK = require( '@stdlib/constants/float32/abs-mask' );
+var FLOAT32_EXPONENT_BIAS = require( '@stdlib/constants/float32/exponent-bias' );
+var FLOAT32_SIGNIFICAND_MASK = require( '@stdlib/constants/float32/significand-mask' );
+var NINF = require( '@stdlib/constants/float32/ninf' );
+var kernelLog1pf = require( '@stdlib/math/base/special/kernel-log1pf' );
+
+
+// VARIABLES //
+
+var TWO25 = f32( 3.3554432000e+07 );   // 0x4c000000
+var IVLN2HI = f32( 1.4428710938e+00 ); /* 0x3fb8b000 */
+var IVLN2LO = f32( -1.7605285393e-04 ); // 0x3de705fc, 0x2eefa200
+var HIGH_MAX_NORMAL_EXP = 0x7f800000|0; // asm type annotation
+var HIGH_MIN_NORMAL_EXP = 0x00800000|0; // asm type annotation
+var HIGH_BIASED_EXP_0 = 0x3f800000|0; // asm type annotation
+
+
+// MAIN //
+
+/**
+* Evaluates the binary logarithm (base two) for single-precision floating point values.
+*
+* @param {NonNegativeNumber} x - input value
+* @returns {number} function value
+*
+* @example
+* var v = log2f( 4.0 );
+* // returns 2.0
+*
+* @example
+* var v = log2f( 8.0 );
+* // returns 3.0
+*
+* @example
+* var v = log2f( 0.0 );
+* // returns -Infinity
+*
+* @example
+* var v = log2f( Infinity );
+* // returns Infinity
+*
+* @example
+* var v = log2f( NaN );
+* // returns NaN
+*
+* @example
+* var v = log2f( -4.0 );
+* // returns NaN
+*/
+function log2f( x ) {
+	var hfsq;
+	var hi;
+	var hx;
+	var lo;
+	var f;
+	var R;
+	var y;
+	var i;
+	var k;
+
+	if ( isnanf( x ) || x < 0.0 ) {
+		return NaN;
+	}
+	if ( x === 0.0 ) {
+		return NINF;
+	}
+	x = f32( x );
+	hx = toWordf( x ) | 0;
+	k = 0|0; // asm type annotation
+	if ( hx < HIGH_MIN_NORMAL_EXP ) {
+		// Case: x < 2**-126
+		if ( hx&ABS_MASK === 0 ) {
+			return NINF;
+		}
+		k -= 25|0; // asm type annotation
+
+		// Subnormal number, scale up x:
+		x = f32( x * TWO25 );
+		hx = toWordf( x ) | 0;
+	}
+	if ( hx >= HIGH_MAX_NORMAL_EXP ) {
+		return f32( x + x );
+	}
+	// Case: log(1) = +0
+	if ( hx === HIGH_BIASED_EXP_0 ) {
+		return 0.0;
+	}
+	k = f32( k + f32( ( ( hx >> 23 ) - FLOAT32_EXPONENT_BIAS ) | 0 ) );
+	hx &= FLOAT32_SIGNIFICAND_MASK;
+	i = f32( ( ( hx+0x4afb0d ) & HIGH_MIN_NORMAL_EXP )|0 ); // asm type annotation
+
+	// Normalize x or x/2...
+	x = fromWordf( hx | ( i ^ HIGH_BIASED_EXP_0) );
+	k = f32( k + ( ( i >> 23 ) | 0 ) );
+	y = f32( k );
+	f = f32( x - 1.0 );
```

Comment:
```suggestion
	f = f32( x - f32( 1.0 ) );
```

Same applies for `0.5` in the next line.